embeddings/data/data_loader.py

```diff
@@ -12,3 +18,42 @@ class DataLoader(ABC, Generic[Input, Output]):
     @abc.abstractmethod
     def load(self, dataset: Dataset[Input]) -> Output:
         pass
+
+
+class HuggingFaceDataLoader(DataLoader[str, datasets.DatasetDict]):
+    def load(self, dataset: Dataset[str]) -> datasets.DatasetDict:
+        if isinstance(dataset, HuggingFaceDataset):
```

Comment:
we can not accept HuggingFaceDataset directly in the method?

Reply:
HuggingFaceDataset is our own class with attributes for the dataset name/path and kwargs used for loading not a HF dataset object.
`class HuggingFaceDataset(Dataset[str]):
    def __init__(self, dataset: Union[str, Path], **load_dataset_kwargs: Any):
        super().__init__()
        self.dataset = dataset
        self.load_dataset_kwargs = load_dataset_kwargs
`
This was only moved from embeddings/data/hugging_face_data_loader.py file and not modified by me.